### PR TITLE
Use numerically stable logp for BG/NBD

### LINKED ISSUE #98: Use numerically stable log-likelihood for `BetaGeoModel`
As per @ColtAllen's [comment](https://github.com/pymc-labs/pymc-marketing/pull/67#discussion_r1042404877), a reformulation to the log-likelihood of the BG/NBD model, added in PR #67, would yield more numerically stable computations (see equation (4) of [this write-up](http://brucehardie.com/notes/027/bgnbd_num_error.pdf)).

### PR DESCRIPTION
**Pull request by larryshamalama**
Closes #98

In the initial PR #67 that adds the BG/NBD model, I ran into some failing test with this "new" logp expression (c.f. [relevant comment](https://github.com/pymc-labs/pymc-marketing/pull/67#discussion_r1042795130)). Tests seem to be passing now?

### CODE REVIEW

**Comment by ColtAllen** [comment]
Why is this only happening with Python 3.10 in the CI?

```
_______________ ERROR collecting tests/clv/test_distributions.py _______________
ImportError while importing test module '/home/runner/work/pymc-marketing/pymc-marketing/tests/clv/test_distributions.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/clv/test_distributions.py:7: in <module>
    from pymc.tests.helpers import select_by_precision
E   ModuleNotFoundError: No module named 'pymc.tests'
```

**Comment by ColtAllen** [comment]
Can we add a test to ensure `logp` matches the results in the paper?

```
x = 200
t_x = 38
T = 40
r = 0.25
alpha = 4.00
a = 0.80
b = 2.50

assert logp == 100.7957
```

**Comment by ricardoV94** [comment]
The missing tests happens in the latest version of PyMC, we will fix soon to make the test utilities accessible again for downstream packages

**Comment by larryshamalama** [comment]
Rebasing following the recently merged #192. More tests should be added

**Comment by larryshamalama** [comment]
Ready for review, the `logp` test was added. I'm getting some inadvertent edits to some other files, probably due to linting?

File: tests/clv/models/test_beta_geo.py [hunk 1]
```diff
@@ -83,6 +83,32 @@ def test_model(self, a_prior, b_prior, alpha_prior, r_prior):
...
+        vv_to_test_values = {
+            vv: value
+            for (vv, value) in zip(
+                pymc_model.value_vars, np.log([0.80, 2.50, 4.00, 0.25])
+            )
+        }
+
+        np.testing.assert_almost_equal(
+            logp.eval(vv_to_test_values), 100.7957, decimal=5
+        )
```
[the remark this answers is not in the document]
Comment (larryshamalama) [reply tests/clv/models/test_beta_geo.py:110]: Sounds like a good idea. Let me incorporate that

**Review by ricardoV94 — APPROVED** [review]
Looks great!

**Pull request merged**